File: aplicacaomvc/Deloitte_NF.Web/Models/NotaFiscal.cs
using System;
using System.Collections.Generic;
using System.Web.Mvc;

namespace Deloitte_NF.Models
{
    /// <summary>
    /// Classe Nota Fiscal
    /// </summary>
    public class NotaFiscal
    {
        /// <summary>
        /// Construtor da classe
        /// </summary>
        public NotaFiscal()
        {
            Cliente = new Cliente();
            Fornecedor = new Fornecedor();
            Produto = new Produto();
        }

        /// <summary>
        /// Obtém e informa o Sequencial da Nota Fiscal
        /// </summary>
        /// <value>Sequencial da Nota Fiscal</value>
        public int IdNotaFiscal { get; set; }

        /// <summary>
        /// Obtém e informa o Cliente da Nota Fiscal
        /// </summary>
        /// <value>Cliente da Nota Fiscal</value>
        public Cliente Cliente { get; set; }

        /// <summary>
        /// Obtém e informa o Produto da Nota Fiscal
        /// </summary>
        /// <value>Produto da Nota Fiscal</value>
        public Produto Produto { get; set; }

        /// <summary>
        /// Obtém e informa o Fornecedor da Nota Fiscal
        /// </summary>
        /// <value>Fornecedor da Nota Fiscal</value>
        public Fornecedor Fornecedor { get; set; }

        /// <summary>
        /// Obtém e informa a Quantidade de Produtos da Nota Fiscal
        /// </summary>
        /// <value>Quantidade de Produtos da Nota Fiscal</value>
        public int QtdeProdutos { get; set; }

        /// <summary>
        /// Obtém e informa a Data da Nota fiscal
        /// </summary>
        /// <value>Data da Nota fiscal</value>
        public DateTime DataNotaFiscal { get; set; }

        /// <summary>
        /// Obtém e informa o Valor da Nota fiscal
        /// </summary>
        /// <value>Valor da Nota fiscal</value>
        public double ValorNotaFiscal { get; set; }

        public IEnumerable<SelectListItem> ListaClientes { get; set; }

        public IEnumerable<SelectListItem> ListaProdutos { get; set; }

        public IEnumerable<SelectListItem> ListaFornecedor { get; set; }
    }
}

File: aplicacaowebforms/Negocio/Entidades/NotaFiscal.vb
Public Class NotaFiscal

    Private _idNotaFiscal As Short
    Private _cliente As Cliente
    Private _produto As Produto
    Private _fornecedor As Fornecedor
    Private _qtdeProdutos As Short
    Private _dataNotaFiscal As DateTime
    Private _valorNotaFiscal As Double

    Public Property IdNotaFiscal As Short
        Get
            Return _idNotaFiscal
        End Get
        Set(value As Short)
            _idNotaFiscal = value
        End Set
    End Property

    Public Property Cliente As Cliente
        Get
            Return _cliente
        End Get
        Set(value As Cliente)
            _cliente = value
        End Set
    End Property

    Public Property Produto As Produto
        Get
            Return _produto
        End Get
        Set(value As Produto)
            _produto = value
        End Set
    End Property

    Public Property Fornecedor As Fornecedor
        Get
            Return _fornecedor
        End Get
        Set(value As Fornecedor)
            _fornecedor = value
        End Set
    End Property

    Public Property QtdeProdutos As Short
        Get
            Return _qtdeProdutos
        End Get
        Set(value As Short)
            _qtdeProdutos = value
        End Set
    End Property

    Public Property DataNotaFiscal As Date
        Get
            Return _dataNotaFiscal
        End Get
        Set(value As Date)
            _dataNotaFiscal = value
        End Set
    End Property

    Public Property ValorNotaFiscal As Double
        Get
            Return _valorNotaFiscal
        End Get
        Set(value As Double)
            _valorNotaFiscal = value
        End Set
    End Property

    ''' <summary>
    ''' Método de listagem de notas fiscais
    ''' </summary>
    ''' <returns>Lista das notas fiscais</returns>
    Public Function ListarNota() As DataTable

        Dim notaFiscalSql As NotaFiscalSQL = New NotaFiscalSQL()

        Return notaFiscalSql.ListarNota()

    End Function

    ''' <summary>
    ''' Método de exclusão de nota fiscal
    ''' </summary>
    Public Sub ExcluirNota()

        Dim notaFiscalSql As NotaFiscalSQL = New NotaFiscalSQL()

        notaFiscalSql.ExcluirNota(Me)

    End Sub

    ''' <summary>
    ''' Método de listagem de clientes
    ''' </summary>
    ''' <returns>Lista de Clientes</returns>
    Public Function ListarClientes() As List(Of Cliente)

        Dim notaFiscalSql As NotaFiscalSQL = New NotaFiscalSQL()

        Return notaFiscalSql.ListarClientes()

    End Function

    ''' <summary>
    ''' Método de listagem de Fornecedores
    ''' </summary>
    ''' <returns>Lista de Fornecedores</returns>
    Public Function ListarFornecedores() As List(Of Fornecedor)

        Dim notaFiscalSql As NotaFiscalSQL = New NotaFiscalSQL()

        Return notaFiscalSql.ListarFornecedores()

    End Function

    ''' <summary>
    ''' Método de listagem de Produto
    ''' </summary>
    ''' <returns>Lista de Produtos</returns>
    Public Function ListarProdutos() As List(Of Produto)

        Dim notaFiscalSql As NotaFiscalSQL = New NotaFiscalSQL()

        Return notaFiscalSql.ListarProdutos()

    End Function

    ''' <summary>
    ''' Método de inclusão de Nota Fiscal
    ''' </summary>
    Public Sub IncluirNota()

        Dim notaFiscalSql As NotaFiscalSQL = New NotaFiscalSQL()

        notaFiscalSql.IncluirNota(Me)

    End Sub

    ''' <summary>
    ''' Método de obtenção de nota fiscal por Identificador
    ''' </summary>
    ''' <returns>Nota fiscal pesquisada</returns>
    Public Function ObterNota() As NameValueCollection

        Dim notaFiscalSql As NotaFiscalSQL = New NotaFiscalSQL()

        Return notaFiscalSql.ObterNota(Me)

    End Function

    ''' <summary>
    ''' Método de atualização de Nota Fiscal
    ''' </summary>
    Public Sub AtualizarNota()

        Dim notaFiscalSql As NotaFiscalSQL = New NotaFiscalSQL()

        notaFiscalSql.AtualizarNota(Me)

    End Sub

    ''' <summary>
    ''' Método de listagem de Notas Fiscais por Produto
    ''' </summary>
    ''' <returns>Listagem de Notas Fiscais por Produto</returns>
    Public Function ListarNotasPorProduto() As DataTable

        Dim notaFiscalSql As New NotaFiscalSQL

        Return notaFiscalSql.ListarNotasPorProduto(Me)

    End Function

End Class